src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2D.java

```diff
@@ -804,4 +845,14 @@ public FontRenderContext getFontRenderContext() {
 		calcGfx.addRenderingHints(renderingHints);
 		return calcGfx.getFontRenderContext();
 	}
+
+	// In case a single 2D graphic object has to be written (one or more times)
+	// into a PDF document, this method isn't needed.
+	// If, instead, several 2D graphic objects have to be written (one or more times)
+	// into a PDF document, the graphic state has to be saved after a graphic object
+	// has been written for the last time and a new graphic object has to be created
+	public void saveGraphicsState() throws IOException
```

Comment:
Is creating a new PdfBoxGraphics2D per chart an option?  That should work and would be the clean way. It should not really need that more space in the final PDF.
At least thats the way I do my charts.

I'll try to do a release later today when I've merged your changes.